datanommer.models/datanommer/models/__init__.py

```diff
@@ -283,15 +297,19 @@ def get_or_create(cls, name):
         Return the instance of the class with the specified name. If it doesn't
         already exist, create it.
         """
-        # Use an INSERT ... SELECT to guarantee we don't get unique constraint
-        # violations if multiple instances of datanommer are trying to insert the same
-        # value at the same time.
-        not_exists = ~exists(select([cls.__table__.c.name]).where(cls.name == name))
-        insert = cls.__table__.insert(inline=True).\
-                 from_select([cls.__table__.c.name],
-                             select([literal(name)]).where(not_exists))
-        session.execute(insert)
-        return cls.query.filter_by(name=name).one()
+        obj = cls.query.filter_by(name=name).one_or_none()
```

Comment:
Note that the ``one_or_none`` method is not in every version of sqlalchemy, so we may want to set this lower version limit somewhere.

Reply:
Updated the sqlalchemy requirement to 1.0.9.